source/Engine.cpp

```diff
@@ -1414,7 +1414,7 @@ void Engine::CalculateStep()
 
 	// Step the weather.
 	for(Weather &weather : activeWeather)
-		weather.Step(newVisuals);
+		weather.Step(newVisuals, flagship);
```

Comment:
This should be
```suggestion
		weather.Step(newVisuals, &flagship->Position());
```
with the assumption that `flagship` is not a null pointer. I haven't checked the surroundings to validate that assumption.

Reply:
I was sure it was checked before... weird that it isn't because when there is no flagship there is no point running anything

Reply:
It'll be null when the flagship dies for example .

Reply:
> It'll be null when the flagship dies for example .

exactly, at which point the game stops